js_modules/dagster-ui/packages/ui-core/src/workspace/WorkspaceContext/WorkspaceContext.tsx

```diff
@@ -281,18 +289,20 @@ async function refreshLocationsIfNeeded(params: RefreshLocationsIfNeededParams):
   }
 
   await Promise.all(
-    locationsToFetch.map((location) =>
-      fetchLocationData({
+    locationsToFetch.map(async (location) => {
+      await fetchLocationData({
         name: location.name,
         client,
         localCacheIdPrefix,
         getData,
         setLocationEntryData,
-      }),
-    ),
+        locationStatuses,
+        fetchingStatusesRef,
+      });
+      previousLocationVersionsRef.current[location.name] =
+        locationStatuses[location.name]?.versionKey ?? '';
```

Comment:
[1] I assume its fine since i see some bits of existing code that do it - but what is the expected result of using empty string when we get null ?

Reply:
Is that a scenario under test or worth putting under test? 

I don't have a strong mental model of all the mechanisms here so I'm just reacting to generally suspicious looking things

Reply:
I moved the version update into the `fetchLocationData` function instead. And instead of using the versionKey from the status update that triggered the fetch, I'm using the versionKey from the location response now so it should mitigate the weirdness here.

Reply:
But yeah, there was a theoretical issue here if the location query triggered by the `refetch` call was slower than a subsequent query... though actually... that issue can still happen... so maybe I should also check if the updateTimestamp is greater than the last version? ugh

Reply:
Random idea: But instead of polling for version keys on the code location status query, maybe I only use that query to get all of the locations, but separately poll the individual locations with a new query that takes the versionKey I'm currently on as an input and either returns the updated code location or a response indicating that I'm already up to date. Would that be easy to do? Could help eliminate some of the complexity here.